File: backend/app/schemas/checkout.py
from pydantic import BaseModel
from decimal import Decimal

from pydantic import BaseModel


class CheckoutResponse(BaseModel):
    order_id: int

    payment_id: int

    subtotal: Decimal

    shipping_cost: Decimal

    discount: Decimal

    total: Decimal

    message: str

    

class CheckoutRequest(BaseModel):
    user_id: int

    address_id: int

    shipping_method_id: int

    payment_method: str

    coupon_code: str | None = None

    notes: str | None = None

File: apps/store/src/lib/api/checkout.ts
import { apiClient } from "./client";

export type CheckoutRequest = {
  user_id: number;
  address_id: number;
  shipping_method_id: number;
  payment_method: string;
  coupon_code?: string | null;
  notes?: string | null;
};

export async function createCheckout(
  data: CheckoutRequest
) {
  return apiClient("/checkout/", {
    method: "POST",
    body: data,
  });
}